pandas/core/indexes/api.py

```diff
@@ -204,7 +204,16 @@ def conv(i):
                 i = i.tolist()
             return i
 
-        return Index(lib.fast_unique_multiple_list([conv(i) for i in inds], sort=sort))
+        # GH 35092. Preserve argument type. This function gets called only when
+        # there is just one type
+        if type(inds[0]) != list:
+            ind_class = type(inds[0])
+        else:
+            ind_class = Index
+
+        return ind_class(
+            lib.fast_unique_multiple_list([conv(i) for i in inds], sort=sort)
+        )
```

Comment:
If this is to work, we need to preserve the argument type, otherwise Index subclasses turn into Index, and everything breaks.

Reply:
Changed the approach completely.